Model/Notifications.php

```diff
@@ -168,14 +177,11 @@ public function proccessNotificatonResult($resultXML, $_payment = false)
             if ((int)$resultXML->status == 3) { 
                 if(!$order->hasInvoices()){
 
-                    $invoice = $order->prepareInvoice();
-                    $invoice->register()->pay();
+                    $invoice = $this->invoiceService->prepareInvoice($order);
                     $msg = sprintf('Captured payment. Transaction Identifier: %s', (string)$resultXML->code);
                     $invoice->addComment($msg);
-                    $invoice->sendEmail(
-                        $this->pagSeguroHelper->getStoreConfigValue('payment/rm_pagseguro/send_invoice_email'),
-                        'Payment received successfully.'
-                    );
+                    $invoice->setRequestedCaptureCase(\RicardoMartins\PagSeguro\Model\Notifications::processStatus($resultXML->status));
```

Comment:
Isn't \RicardoMartins\PagSeguro\Model\Notifications same as $this ?

Reply:
I'm also afraid this is not what setRequestedCaptureCase is expecting, neither if we should use it here. It seems its supposed to determine if the Invoice model will capture or not the order, but our order was captured already when we first sent the request to pagseguro.
See vendor/magento/magento2-base/dev/tests/integration/testsuite/Magento/Sales/_files/invoice_payflowpro.php